--- gcc-10.2.0/config.sub	2020-09-28 15:08:06.829807463 +0200
+++ config.sub.orig	2020-09-28 15:08:20.883111701 +0200
@@ -1366,7 +1366,7 @@ case $os in
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
-	     | nsk* | powerunix)
+	     | nsk* | powerunix | xelix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
--- gcc-10.2.0/fixincludes/mkfixinc.sh
+++ b/fixincludes/mkfixinc.sh
@@ -11,6 +11,7 @@ target=fixinc.sh

 # Check for special fix rules for particular targets
 case $machine in
+    *-xelix-* | \
     i?86-*-cygwin* | \
     i?86-*-mingw32* | \
     x86_64-*-mingw32* | \
--- gcc-10.2.0/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -935,6 +935,13 @@ case ${target} in
   default_use_cxa_atexit=yes
   use_gcc_stdint=wrap
   ;;
+*-*-xelix*)
+  extra_parts="crtbegin.o crtend.o"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  use_gcc_stdint=provide
+  ;;
 esac

 case ${target} in
@@ -1542,6 +1548,9 @@ x86_64-*-openbsd*)
 	gas=yes
 	gnu_ld=yes
 	;;
+i[34567]86-*-*)
+    tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h xelix.h"
+    ;;
 i[34567]86-*-linux* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-gnu* | i[34567]86-*-kopensolaris*-gnu)
 			# Intel 80386's running GNU/*
 			# with ELF format using glibc 2
--- gcc-10.2.0/libgcc/config.host
+++ b/libgcc/config.host
@@ -753,6 +753,10 @@ i[34567]86-*-mingw*)
 	fi
 	tmake_file="${tmake_file} ${tmake_eh_file} ${tmake_dlldir_file} i386/t-slibgcc-cygming i386/t-cygming i386/t-mingw32 t-crtfm i386/t-chkstk t-dfprules"
 	;;
+i[34567]86-*-xelix*)
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o crtbeginS.o crtendS.o"
+	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
 x86_64-*-mingw*)
 	case ${target_thread_file} in
 	  win32)
--- gcc-10.2.0/libstdc++-v3/crossconfig.m4	2019-02-28 00:22:34.907419609 +0100
+++ gcc-10.2.0/libstdc++-v3/crossconfig.m4	2019-02-28 00:23:40.386423732 +0100
@@ -291,6 +291,12 @@
     AC_DEFINE(HAVE_TANF)
     AC_DEFINE(HAVE_TANHF)
     ;;
+  *-xelix)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_LINKER_FEATURES
+    GLIBCXX_CHECK_MATH_SUPPORT
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;
   *)
     AC_MSG_ERROR([No support for this host/target combination.])
    ;;
--- gcc-10.2.0/libstdc++-v3/configure.host	2019-02-22 08:25:52.000000000 +0100
+++ gcc-10.2.0/libstdc++-v3/configure.host	2020-01-07 01:48:34.684537249 +0100
@@ -318,6 +318,9 @@ case "${host_os}" in
   vxworks)
     os_include_dir="os/vxworks"
     ;;
+  xelix)
+    os_include_dir="os/generic"
+    ;;
   *)
     os_include_dir="os/generic"
     ;;
--- gcc-10.2.0/libstdc++-v3/configure	2020-09-30 00:07:36.453707692 +0200
+++ gcc-10.2.0/libstdc++-v3/configure	2020-09-30 00:07:17.023791569 +0200
@@ -74117,9 +74117,9 @@ _ACEOF


     ;;
-  *)
-    as_fn_error $? "No support for this host/target combination." "$LINENO" 5
-   ;;
+#  *)
+#    as_fn_error $? "No support for this host/target combination." "$LINENO" 5
+#   ;;
 esac

   fi
--- gcc-10.2.0/gcc/configure	2020-09-30 01:02:35.104945210 +0200
+++ gcc-10.2.0/gcc/configure	2020-09-30 01:02:45.814901591 +0200
@@ -11623,13 +11623,6 @@ rlim_t l = 0;
   return 0;
 }
 _ACEOF
-if ac_fn_cxx_try_compile "$LINENO"; then :
-
-else
-
-$as_echo "#define rlim_t long" >>confdefs.h
-
-fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext

 # On AIX 5.2, <ldfcn.h> conflicts with <fcntl.h>, as both define incompatible
