--- coreutils-8.30/Makefile	2019-02-04 20:37:41.509799534 +0100
+++ coreutils-8.30/Makefile	2019-02-04 02:52:58.585407515 +0100
@@ -13896,26 +13896,7 @@
 # We need the following in order to create <sys/select.h> when the system
 # doesn't have one that works with the given compiler.
 lib/sys/select.h: lib/sys_select.in.h $(top_builddir)/config.status $(CXXDEFS_H) $(WARN_ON_USE_H)
-	$(AM_V_at)$(MKDIR_P) lib/sys
-	$(AM_V_GEN)rm -f $@-t $@ && \
-	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
-	  sed -e 's|@''GUARD_PREFIX''@|GL|g' \
-	      -e 's|@''INCLUDE_NEXT''@|$(INCLUDE_NEXT)|g' \
-	      -e 's|@''PRAGMA_SYSTEM_HEADER''@|#pragma GCC system_header|g' \
-	      -e 's|@''PRAGMA_COLUMNS''@||g' \
-	      -e 's|@''NEXT_SYS_SELECT_H''@|$(NEXT_SYS_SELECT_H)|g' \
-	      -e 's|@''HAVE_SYS_SELECT_H''@|$(HAVE_SYS_SELECT_H)|g' \
-	      -e 's/@''GNULIB_PSELECT''@/$(GNULIB_PSELECT)/g' \
-	      -e 's/@''GNULIB_SELECT''@/$(GNULIB_SELECT)/g' \
-	      -e 's|@''HAVE_WINSOCK2_H''@|$(HAVE_WINSOCK2_H)|g' \
-	      -e 's|@''HAVE_PSELECT''@|$(HAVE_PSELECT)|g' \
-	      -e 's|@''REPLACE_PSELECT''@|$(REPLACE_PSELECT)|g' \
-	      -e 's|@''REPLACE_SELECT''@|$(REPLACE_SELECT)|g' \
-	      -e '/definitions of _GL_FUNCDECL_RPL/r $(CXXDEFS_H)' \
-	      -e '/definition of _GL_WARN_ON_USE/r $(WARN_ON_USE_H)' \
-	      < $(top_srcdir)/lib/sys_select.in.h; \
-	} > $@-t && \
-	mv $@-t $@
+	echo

 # We need the following in order to create <sys/socket.h> when the system
 # doesn't have one that works with the given compiler.
--- coreutils-8.30/lib/unistd.h	2019-02-04 21:02:21.718080262 +0100
+++ coreutils-8.30-patched/lib/unistd.h	2019-02-04 07:36:14.662698556 +0100
@@ -1305,86 +1305,7 @@
 #endif


-#if 1
-# if 0
-#  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
-#   define getpagesize rpl_getpagesize
-#  endif
-_GL_FUNCDECL_RPL (getpagesize, int, (void));
-_GL_CXXALIAS_RPL (getpagesize, int, (void));
-# else
-#  if !0
-#   if !defined getpagesize
-/* This is for POSIX systems.  */
-#    if !defined _gl_getpagesize && defined _SC_PAGESIZE
-#     if ! (defined __VMS && __VMS_VER < 70000000)
-#      define _gl_getpagesize() sysconf (_SC_PAGESIZE)
-#     endif
-#    endif
-/* This is for older VMS.  */
-#    if !defined _gl_getpagesize && defined __VMS
-#     ifdef __ALPHA
-#      define _gl_getpagesize() 8192
-#     else
-#      define _gl_getpagesize() 512
-#     endif
-#    endif
-/* This is for BeOS.  */
-#    if !defined _gl_getpagesize && 0
-#     include <OS.h>
-#     if defined B_PAGE_SIZE
-#      define _gl_getpagesize() B_PAGE_SIZE
-#     endif
-#    endif
-/* This is for AmigaOS4.0.  */
-#    if !defined _gl_getpagesize && defined __amigaos4__
-#     define _gl_getpagesize() 2048
-#    endif
-/* This is for older Unix systems.  */
-#    if !defined _gl_getpagesize && 1
-#     include <sys/param.h>
-#     ifdef EXEC_PAGESIZE
-#      define _gl_getpagesize() EXEC_PAGESIZE
-#     else
-#      ifdef NBPG
-#       ifndef CLSIZE
-#        define CLSIZE 1
-#       endif
-#       define _gl_getpagesize() (NBPG * CLSIZE)
-#      else
-#       ifdef NBPC
-#        define _gl_getpagesize() NBPC
-#       endif
-#      endif
-#     endif
-#    endif
-#    if !(defined __cplusplus && defined GNULIB_NAMESPACE)
-#     define getpagesize() _gl_getpagesize ()
-#    else
-#     if !GNULIB_defined_getpagesize_function
-_GL_UNISTD_INLINE int
-getpagesize ()
-{
-  return _gl_getpagesize ();
-}
-#      define GNULIB_defined_getpagesize_function 1
-#     endif
-#    endif
-#   endif
-#  endif
-/* Need to cast, because on Cygwin 1.5.x systems, the return type is size_t.  */
-_GL_CXXALIAS_SYS_CAST (getpagesize, int, (void));
-# endif
-# if 1
-_GL_CXXALIASWARN (getpagesize);
-# endif
-#elif defined GNULIB_POSIXCHECK
-# undef getpagesize
-# if HAVE_RAW_DECL_GETPAGESIZE
-_GL_WARN_ON_USE (getpagesize, "getpagesize is unportable - "
-                 "use gnulib module getpagesize for portability");
-# endif
-#endif
+#define getpagesize() 4096


 #if 1
