diff -rupN download/binutils-2.22/bfd/config.bfd download/binutils-2.22-xelix/bfd/config.bfd
--- download/binutils-2.22/bfd/config.bfd	2011-07-29 00:35:13.000000000 +0200
+++ download/binutils-2.22-xelix/bfd/config.bfd	2012-01-07 04:24:53.000000000 +0100
@@ -141,6 +141,11 @@ case "${targ}" in
     ;;
 
 # START OF targmatch.h
+  i[3-7]86-*-xelix*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selfvecs=i386coff_vec
+    ;;
+
 #ifdef BFD64
   alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu)
     targ_defvec=bfd_elf64_alpha_freebsd_vec
diff -rupN download/binutils-2.22/config.sub download/binutils-2.22-xelix/config.sub
--- download/binutils-2.22/config.sub	2011-06-06 12:36:06.000000000 +0200
+++ download/binutils-2.22-xelix/config.sub	2012-01-07 03:18:37.000000000 +0100
@@ -1317,7 +1317,7 @@ case $os in
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -xelix* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
diff -rupN download/binutils-2.22/gas/configure.tgt download/binutils-2.22-xelix/gas/configure.tgt
--- download/binutils-2.22/gas/configure.tgt	2011-06-20 15:23:21.000000000 +0200
+++ download/binutils-2.22-xelix/gas/configure.tgt	2012-01-07 03:18:37.000000000 +0100
@@ -91,6 +91,7 @@ esac
 generic_target=${cpu_type}-$vendor-$os
 # Note: This table is alpha-sorted, please try to keep it that way.
 case ${generic_target} in
+  i386-*-xelix*)			fmt=elf ;;
   alpha-*-*vms*)			fmt=evax ;;
   alpha-*-osf*)				fmt=ecoff ;;
   alpha-*-linuxecoff*)			fmt=ecoff ;;
diff -rupN download/binutils-2.22/ld/configure.tgt download/binutils-2.22-xelix/ld/configure.tgt
--- download/binutils-2.22/ld/configure.tgt	2011-11-21 10:29:37.000000000 +0100
+++ download/binutils-2.22-xelix/ld/configure.tgt	2012-01-07 03:18:37.000000000 +0100
@@ -27,6 +27,7 @@ targ64_extra_libpath=
 # architecture variants should be kept together even if their names
 # break the alpha sorting.
 case "${targ}" in
+i[3-7]86-*-xelix*) targ_emul=xelix_i386 ;;
 alpha*-*-freebsd* | alpha*-*-kfreebsd*-gnu)
 		        targ_emul=elf64alpha_fbsd
 			targ_extra_emuls="elf64alpha alpha"
diff -rupN download/binutils-2.22/ld/Makefile.in download/binutils-2.22-xelix/ld/Makefile.in
--- download/binutils-2.22/ld/Makefile.in	2011-07-22 22:22:37.000000000 +0200
+++ download/binutils-2.22-xelix/ld/Makefile.in	2012-01-07 03:18:37.000000000 +0100
@@ -2627,6 +2627,9 @@ eelf32xtensa.c: $(srcdir)/emulparams/elf
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386 "$(tdir_elf_i386)"
+exelix_i386.c: $(srcdir)/emulparams/xelix_i386.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} xelix_i386 "$(tdir_xelix_i386)"
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
