--- newlib-3.2.0/config.sub	2020-01-05 02:06:30.961047237 +0100
+++ newlib-3.2.0/config.sub	2020-01-05 02:14:39.620713316 +0100
@@ -1363,7 +1363,7 @@ case $os in
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed* | emscripten*)
+	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | xelix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
--- newlib-3.2.0/newlib/configure.host	2019-02-08 18:12:13.508918038 +0100
+++ newlib-3.2.0/newlib/configure.host	2019-02-08 18:13:05.578570795 +0100
@@ -602,6 +602,11 @@
   w65-*-*)
 	sys_dir=w65
 	;;
+  *-*-xelix*)
+	sys_dir=xelix
+	posix_dir=posix
+	unix_dir=unix
+	;;
   z8k-*-coff)
 	sys_dir=z8ksim
 	;;
@@ -887,6 +892,10 @@
   xc16x-*)
           syscall_dir=syscalls
         ;;
+  *-*-xelix*)
+	syscall_dir=syscalls
+	newlib_cflags="${newlib_cflags} -DHAVE_FCNTL -DHAVE_MMAP"
+	;;
   xstormy16-*-*)
 	syscall_dir=syscalls
 	;;
diff -rupN newlib-3.2.0/newlib/libc/sys/configure.in newlib-3.2.0-xelix/newlib/libc/sys/configure.in
--- newlib-3.2.0/newlib/libc/sys/configure.in	2010-02-24 21:59:55.000000000 +0100
+++ newlib-3.2.0-xelix/newlib/libc/sys/configure.in	2012-01-06 23:38:45.000000000 +0100
@@ -22,6 +22,7 @@ fi

 if test -n "${sys_dir}"; then
   case ${sys_dir} in
+	xelix) AC_CONFIG_SUBDIRS(xelix) ;;
 	a29khif) AC_CONFIG_SUBDIRS(a29khif) ;;
 	arc) AC_CONFIG_SUBDIRS(arc) ;;
 	arm) AC_CONFIG_SUBDIRS(arm) ;;
--- newlib-3.2.0/newlib/libc/include/sys/stat.h	2018-07-23 09:34:43.211471869 +0200
+++ newlib-3.2.0/newlib/libc/include/sys/stat.h	2018-07-23 09:34:37.564479109 +0200
@@ -152,7 +152,7 @@
 int	stat (const char *__restrict __path, struct stat *__restrict __sbuf );
 mode_t	umask (mode_t __mask );

-#if defined (__SPU__) || defined(__rtems__) || defined(__CYGWIN__) && !defined(__INSIDE_CYGWIN__)
+#if defined (__SPU__) || defined(__rtems__) || defined(__xelix__) || defined(__CYGWIN__) && !defined(__INSIDE_CYGWIN__)
 int	lstat (const char *__restrict __path, struct stat *__restrict __buf );
 int	mknod (const char *__path, mode_t __mode, dev_t __dev );
 #endif
--- newlib-3.2.0/newlib/libc/include/glob.h	2018-07-23 10:32:35.198264689 +0200
+++ newlib-3.2.0/newlib/libc/include/glob.h	2018-07-23 10:32:10.619084204 +0200
@@ -74,6 +74,7 @@
 #define	GLOB_QUOTE	0x0400	/* Quote special chars with \. */
 #define	GLOB_TILDE	0x0800	/* Expand tilde names from the passwd file. */
 #define	GLOB_LIMIT	0x1000	/* limit number of returned paths */
+#define GLOB_NOMATCH 0xffff

 /* backwards compatibility, this is the old name for this option */
 #define GLOB_MAXPATH	GLOB_LIMIT

--- newlib-3.2.0/newlib/libc/include/sys/resource.h	2018-07-23 10:36:09.427233216 +0200
+++ newlib-3.2.0/newlib/libc/include/sys/resource.h	2018-07-23 10:35:38.407046458 +0200
@@ -6,11 +6,20 @@
 #define	RUSAGE_SELF	0		/* calling process */
 #define	RUSAGE_CHILDREN	-1		/* terminated child processes */

+#define RLIM_INFINITY 0
+
 struct rusage {
   	struct timeval ru_utime;	/* user time used */
 	struct timeval ru_stime;	/* system time used */
 };

+typedef uint32_t rlim_t;
+
+struct rlimit {
+	rlim_t rlim_cur;  /* Soft limit */
+	rlim_t rlim_max;  /* Hard limit (ceiling for rlim_cur) */
+};
+
 int	getrusage (int, struct rusage*);

 #endif
--- newlib-3.2.0/newlib/libc/stdlib/mallocr.c	2020-01-02 20:56:24.000000000 +0100
+++ newlib-3.2.0/newlib/libc/stdlib/mallocr.c	2023-02-20 22:32:04.738171041 +0100
@@ -305,7 +305,11 @@ extern "C" {
 
 #define POINTER_UINT unsigned _POINTER_INT
 #define SEPARATE_OBJECTS
-#define HAVE_MMAP 0
+
+#ifndef HAVE_MMAP
+# define HAVE_MMAP 0
+#endif
+
 #define MORECORE(size) _sbrk_r(reent_ptr, (size))
 #define MORECORE_CLEARS 0
 #define MALLOC_LOCK __malloc_lock(reent_ptr)
