--- nano-3.1/config.sub	2017-07-06 16:08:33.000000000 +0200
+++ ../../../nano-3.1//config.sub	2018-10-07 17:14:43.112575885 +0200
@@ -1417,7 +1417,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -xelix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
