--- bash-4.4/support/config.sub	2013-12-17 16:49:47.000000000 +0100
+++ ../../../bash-4.4/support/config.sub	2018-10-07 05:02:03.820062850 +0200
@@ -1376,7 +1376,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -xelix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
