--- mpc-1.1.0/config.sub	2016-01-12 18:30:16.000000000 +0100
+++ mpc-1.1.0/config.sub	2019-02-06 17:27:32.733102712 +0100
@@ -1393,7 +1393,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -tirtos*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -tirtos* | -xelix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
