diff -ruPN -x docs picotcp-1.7.0/include/pico_config.h ../../../picotcp-1.7.0/include/pico_config.h
--- picotcp-1.7.0/include/pico_config.h	2017-02-08 11:37:51.000000000 +0100
+++ ../../../picotcp-1.7.0/include/pico_config.h	2018-09-20 18:48:47.729294662 +0200
@@ -227,6 +227,8 @@
 # include "arch/pico_generic_gcc.h"
 #elif defined __KERNEL__
 # include "arch/pico_linux.h"
+#elif defined __xelix__
+# include "arch/pico_xelix.h"
 /* #elif defined ... */
 #else
 # include "arch/pico_posix.h"
diff -ruPN -x docs ../../../picotcp-1.7.0/Makefile picotcp-1.7.0/Makefile
--- ../../../picotcp-1.7.0/Makefile	2017-02-08 11:37:51.000000000 +0100
+++ picotcp-1.7.0/Makefile	2018-09-30 17:18:23.876916741 +0200
@@ -37,6 +37,7 @@
 ICMP4?=1
 MCAST?=1
 DEVLOOP?=1
+DEVXELIX?=1
 PING?=1
 DHCP_CLIENT?=1
 DHCP_SERVER?=1
@@ -263,6 +264,9 @@
 ifneq ($(DEVLOOP),0)
   include rules/devloop.mk
 endif
+ifneq ($(DEVXELIX),0)
+  include rules/devxelix.mk
+endif
 ifneq ($(DHCP_CLIENT),0)
   include rules/dhcp_client.mk
 endif
