diff --git download/binutils-2.26/bfd/config.bfd b/bfd/config.bfd
index c5688cb..37107cc 100644
--- download/binutils-2.26/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -169,6 +169,10 @@ case "${targ}" in
 
 # START OF targmatch.h
 #ifdef BFD64
+  i[3-7]86-*-xelix*)
+    targ_defvec=i386_elf32_vec
+    targ_selfvecs=i386coff_vec
+    ;;
   aarch64-*-elf)
     targ_defvec=aarch64_elf64_le_vec
     targ_selvecs="aarch64_elf64_be_vec aarch64_elf32_le_vec aarch64_elf32_be_vec arm_elf32_le_vec arm_elf32_be_vec"
diff --git download/binutils-2.26/config.sub b/config.sub
index f018151..c10f7da 100755
--- download/binutils-2.26/config.sub
+++ b/config.sub
@@ -1374,7 +1374,7 @@ case $os in
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -xelix* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
diff --git download/binutils-2.26/gas/configure.tgt b/gas/configure.tgt
index 0b490d4..e08aa52 100644
--- download/binutils-2.26/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -120,6 +120,7 @@ esac
 generic_target=${cpu_type}-$vendor-$os
 # Note: This table is alpha-sorted, please try to keep it that way.
 case ${generic_target} in
+  i386-*-xelix*)			fmt=elf ;;
   aarch64*-*-elf)			fmt=elf;;
   aarch64*-*-linux*)			fmt=elf em=linux ;;

diff --git download/binutils-2.26/ld/Makefile.in b/ld/Makefile.in
index ed98f87..30b103f 100644
--- download/binutils-2.26/ld/Makefile.in
+++ b/ld/Makefile.in
@@ -2801,6 +2801,9 @@ eelf32xtensa.c: $(srcdir)/emulparams/elf32xtensa.sh $(ELF_DEPS) \
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+exelix_i386.c: $(srcdir)/emulparams/xelix_i386.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
  
diff --git download/binutils-2.26/ld/configure.tgt b/ld/configure.tgt
index 6b6bbf2..3b10c34 100644
--- download/binutils-2.26/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -45,6 +45,7 @@ targ64_extra_libpath=
 # architecture variants should be kept together even if their names
 # break the alpha sorting.
 case "${targ}" in
+i[3-7]86-*-xelix*)	targ_emul=xelix_i386 ;;
 aarch64_be-*-elf)	targ_emul=aarch64elfb
 			targ_extra_emuls="aarch64elf aarch64elf32 aarch64elf32b armelfb armelf" ;;
 aarch64-*-elf)		targ_emul=aarch64elf
