diff -rupN download/gcc-4.6.2/config.sub download/gcc-4.6.2-xelix/config.sub
--- download/gcc-4.6.2/config.sub	2010-05-25 15:22:07.000000000 +0200
+++ download/gcc-4.6.2-xelix/config.sub	2012-01-06 22:41:24.000000000 +0100
@@ -1292,7 +1292,7 @@ case $os in
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -xelix* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
diff -rupN download/gcc-4.6.2/gcc/config.gcc download/gcc-4.6.2-xelix/gcc/config.gcc
--- download/gcc-4.6.2/gcc/config.gcc	2011-07-22 18:44:50.000000000 +0200
+++ download/gcc-4.6.2-xelix/gcc/config.gcc	2012-01-06 22:50:46.000000000 +0100
@@ -482,6 +482,12 @@ tm_defines="$tm_defines LIBC_GLIBC=1 LIB
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-xelix*)
+  extra_parts="crtbegin.o crtend.o"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  ;;
 *-*-darwin*)
   tmake_file="t-darwin ${cpu_type}/t-darwin"
   tm_file="${tm_file} darwin.h"
@@ -704,6 +710,11 @@ esac
 
 case ${target} in
 # Support site-specific machine types.
+i[3-7]86-*-xelix*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h xelix.h"
+	tmake_file="i386/t-i386elf i386/t-crtstuff t-svr4"
+	use_fixproto=yes
+	;;
 *local*)
 	rest=`echo ${target} | sed -e "s/$cpu_type-//"`
 	tm_file=${cpu_type}/$rest.h
diff -rupN download/gcc-4.6.2/libgcc/config.host download/gcc-4.6.2-xelix/libgcc/config.host
--- download/gcc-4.6.2/libgcc/config.host	2011-03-14 07:06:23.000000000 +0100
+++ download/gcc-4.6.2-xelix/libgcc/config.host	2012-01-06 22:52:22.000000000 +0100
@@ -176,6 +176,8 @@ esac
 
 case ${host} in
 # Support site-specific machine types.
+i[3-7]86-*-xelix*)
+	;;
 *local*)
 	rest=`echo ${host} | sed -e "s/$cpu_type-//"`
 	if test -f $srcdir/config/${cpu_type}/t-$rest

diff -rupN download/gcc-4.6.2/gcc/gengtype.c download/gcc-4.6.2-xelix/gcc/gengtype.c
index abf17f8..550d3bb 100644
--- download/gcc-4.6.2/gcc/gengtype.c
+++ download/gcc-4.6.2-xelix/gcc/gengtype.c
@@ -3594,13 +3594,13 @@ write_field_root (outf_p f, pair_p v, type_p type, const char *name,
 		  int has_length, struct fileloc *line, const char *if_marked,
 		  bool emit_pch, type_p field_type, const char *field_name)
 {
+  struct pair newv;
   /* If the field reference is relative to V, rather than to some
      subcomponent of V, we can mark any subarrays with a single stride.
      We're effectively treating the field as a global variable in its
      own right.  */
   if (v && type == v->type)
     {
-      struct pair newv;
 
       newv = *v;
       newv.type = field_type;
