diff -rupN download/newlib-2.2.0.20151023/config.sub download/newlib-2.2.0.20151023-xelix/config.sub
--- download/newlib-2.2.0.20151023/config.sub	2011-11-02 01:56:53.000000000 +0100
+++ download/newlib-2.2.0.20151023-xelix/config.sub	2012-01-06 23:35:49.000000000 +0100
@@ -1328,7 +1328,7 @@ case $os in
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -xelix* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
diff -rupN download/newlib-2.2.0.20151023/newlib/configure.host download/newlib-2.2.0.20151023-xelix/newlib/configure.host
--- download/newlib-2.2.0.20151023/newlib/configure.host	2011-11-29 07:33:48.000000000 +0100
+++ download/newlib-2.2.0.20151023-xelix/newlib/configure.host	2012-01-06 23:38:03.000000000 +0100
@@ -343,6 +343,9 @@ esac
 # THIS TABLE IS ALPHA SORTED.  KEEP IT THAT WAY.
 
 case "${host}" in
+  i[3-7]86-*-xelix*)
+	sys_dir=xelix
+	;;
   *-*-cygwin*)
 	posix_dir=posix
 	stdio64_dir=stdio64
diff -rupN download/newlib-2.2.0.20151023/newlib/libc/sys/configure.in download/newlib-2.2.0.20151023-xelix/newlib/libc/sys/configure.in
--- download/newlib-2.2.0.20151023/newlib/libc/sys/configure.in	2010-02-24 21:59:55.000000000 +0100
+++ download/newlib-2.2.0.20151023-xelix/newlib/libc/sys/configure.in	2012-01-06 23:38:45.000000000 +0100
@@ -22,6 +22,7 @@ fi
 
 if test -n "${sys_dir}"; then
   case ${sys_dir} in
+	xelix) AC_CONFIG_SUBDIRS(xelix) ;;
 	a29khif) AC_CONFIG_SUBDIRS(a29khif) ;;
 	arc) AC_CONFIG_SUBDIRS(arc) ;;
 	arm) AC_CONFIG_SUBDIRS(arm) ;;
